app/services/credits/progressive_billing_service.rb

```diff
@@ -58,17 +58,10 @@ def call
 
     attr_reader :invoice
 
-    def apply_credit_to_fees(credit, total_charges_amount)
-      base_amount = total_charges_amount - invoice.progressive_billing_credit_amount_cents
-
-      invoice.fees.charge.reload.each do |fee|
-        if base_amount.positive?
-          fee.precise_coupons_amount_cents += (
-            credit.amount_cents * (fee.amount_cents - fee.precise_coupons_amount_cents)
-          ).fdiv(base_amount)
-        end
-
-        fee.precise_coupons_amount_cents = fee.amount_cents if fee.amount_cents < fee.precise_coupons_amount_cents
+    def apply_credit_to_fees(progressive_billing_invoice)
+      progressive_billing_invoice.fees.charge.each do |progressive_fee|
+        fee = invoice.fees.find_by(charge_id: progressive_fee.charge_id)
```

Comment:
You should also filter by `charge_filter_id` and `grouped_by` as multiple fees could share the same `charge_id`